rust_dev_preview/cross_service/rest_ses/src/work_item/collection.rs

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Provides scoped HTTP endpoints for a REST WorkItem collection.
+//! This includes the common REST HTTP endpoints and also RPC-like endpoints.
+//!
+//! * `GET /items/` for list.
+//! * `GET /items/{itemid} to retrieve.
+//! * `POST /items/` for create.
+//! * `PUT /items/{itemid}` to update.
+//! * `DELETE /items/{itemid}` to delete.
+//! * `PUT /items/{itemid}:archive` to mark an item as archived.
+
+use actix_web::{
+    get, post, put,
+    web::{self, Data, Json, Path, Query},
+    Scope,
+};
+
+use super::{WorkItem, WorkItemArchived, WorkItemError};
+use crate::client::RdsClient;
+
+/// Create the root collection scope.
+pub fn scope() -> Scope {
+    web::scope("/items")
+        .service(create)
+        .service(retrieve)
+        .service(list)
+        .service(update)
+        .service(delete)
+        .service(archive)
+}
+
+/// Create a single WorkItem, serialized as a JSON body.
+#[post("")]
+#[tracing::instrument(
+        name = "Request Create new WorkItem",
+        skip(item, client),
+        fields(work_item.user = %item.name, work_item.guide = %item.guide,)
+    )]
+async fn create(
+    item: Json<WorkItem>,
+    client: Data<RdsClient>,
+) -> Result<Json<WorkItem>, WorkItemError> {
+    super::repository::create(item.0, &client).await.map(Json)
+}
+
+/// Retrieve a single WorkItem, in a JSON body, specified by a URL parameter.
+#[get("/{id}")]
+#[tracing::instrument(name = "Request Retrieve single WorkItem", skip(client))]
+async fn retrieve(
+    itemid: Path<String>,
+    client: Data<RdsClient>,
+) -> Result<Json<WorkItem>, WorkItemError> {
+    super::repository::retrieve(itemid.to_string(), &client)
+        .await
+        .map(Json)
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct ListParams {
+    archived: Option<bool>,
+}
+
+/// List all WorkItems, with an optional archived query parameter.
+/// The archived parameter defaults to active items.
```

Comment:
My Rust is weak, but the spec says to return all items when `archived` is not present, and my read is that you're returning active items in that case.

Reply:
Good catch, `None` in the match arm is a specific case rather than a default case.